llvm/test/tools/repo2obj/extern_weak_symbol.ll

```diff
@@ -0,0 +1,70 @@
+; Check the weak symbol binding.
+;
+; Case 1: the output symbol is weak if all of the references to it are weak.
+; RUN: rm -f %t.db
+; RUN: env REPOFILE=%t.db llc -filetype=obj %s -o %t.o
+; RUN: pstore-dump --all-fragments %t.db | FileCheck -check-prefix=MC-WEAK %s
+; RUN: repo2obj %t.o --repo  %t.db -o %t.elf
+; RUN: llvm-readobj --symbols %t.elf | FileCheck %s
+;
+; Case 2: One or more strong references in the input produces a strong reference in the output.
+; RUN: pstore-export %t.db > %t.json
+; RUN: %python %S/Inputs/modify_json.py -i %t.json -g _ZTH1a -o %t1.json
```

Comment:
I’m not keen on this approach. It would be good to decouple the repo2obj tests from the compiler where possible so that we have total control over the input and aren’t subject to test behaviour changing when the compiler is updated.

I’d suggest using the output from pstore-export as the source for this test (c.f. rld/test/phdr.json). I’d also suggest splitting the two cases into two different tests to a) simplify the tests and b) make it more obvious what has failed. To understand the inputs to the test I first have to understand exactly what modify_json.py is doing. That’s a bit of barrier to easily understanding a test failure!

Reply:
Thanks for your suggestion. I have updated the test.